wordpress-mobile/WordPress-Android
# [Jetpack Social] Implement "Not now" button in post settings no connections container

Part of #18571

Should only be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/18869

To test:
1 - Enable JetpackSocialFeatureConfig feature flag;
2 - Make sure you have **no** sharing accounts connected;
3 - Select a SH site with Jetpack plugin or Jetpack Social plugin installed;
4 - Open post list;
5 - Select a draft or create a new one;
6 - Tap the overflow menu and select "Post settings";
7 - Scroll down to the "Social Sharing" section;
9 - The no connections UI should be shown, with the buttons `"Connect accounts"` and `"Not now"`;
10 - Tap `"Not now"`: the "Social Sharing" container should be gone;
11 - Close the screen and open it again: the container should still be gone;
12 - Add some social account using `My Site` -> `Menu` -> `Social` and open post settings again: the "Social Sharing" loaded state should be shown;
13 - Remove the social connection and open post settings again: no "Social Sharing" section should be shown, including the no connections UI.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and updated unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated `EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt` and `EditPostPublishSettingsViewModelTest.kt`

PR submission checklist:


UI Changes testing checklist:
